main/unbound/unbound.initd

```diff
@@ -22,36 +22,22 @@ depend() {
 	after auth-dns
 }
 
-checkconfig() {
-	checkpath --directory /var/run/unbound
-	UNBOUND_PIDFILE=$("${UNBOUND_CHECKCONF}" -o pidfile "${UNBOUND_CONFFILE}")
-	return $?
-}
-
 configtest() {
-	ebegin "Checking ${SVCNAME} configuration"
-	checkconfig
-	eend $?
-}
-
-start() {
-	checkconfig || return $?
-	ebegin "Starting ${SVCNAME}"
-	start-stop-daemon --start --pidfile "${UNBOUND_PIDFILE}" \
-		--exec "${UNBOUND_BINARY}" -- -c "${UNBOUND_CONFFILE}"
+	ebegin "Checking $RC_SVCNAME configuration"
+	# unbound uses the pid in $cfgfile & we set it via initd
+	sed -i "s|.*pidfile:.*|pidfile:  \"$pidfile\"|" "$cfgfile"
```

Comment:
We cannot have init.d script modify users configuration files.

We should extract the pidfile from unbounds config (like current init.d script does) and use that instead of hardcode pidfile in init.d script and rewrite users config.

Reply:
yes I agree - the original form of my PR which simply added support to turn on `debug` logging is all that should be applied - not hacks like this so the `pid` is set in the `initd`

Reply:
The best option would be to start unbound in foreground, don't let it create pidfile and let init system do its job.

Reply:
any reason why this is still not pulled ???

Reply:
It’s a wrong approach.

> The best option would be to start unbound in foreground, don't let it create pidfile and let init system do its job.

Reply:
if you start it in the foreground the `initd` never exits.

if anything the `-d` needs to be removed as running in the foreground breaks the `initd`

Reply:
`-d` removed

Reply:
`command_background="yes"`. @Shizmob has already mentioned it few comments above.

Reply:
yes it has `command_background="yes"` - so setting `-d` has zero effect so I removed the `-d`

Reply:
![broken-unbound](https://it-offshore.co.uk/downloads/broken-unbound.gif)

broken unbound `initd`

Reply:
![working-pr-unbound](https://it-offshore.co.uk/downloads/working-pr-unbound.gif)

working unbound `initd` running in background as in the `PR`

Reply:
This is a misunderstanding.

> The best option would be to **start unbound in foreground**…

This means start the unbound program in foreground, i.e. do not let it daemonize itself, i.e. start it without `-d` option.

>> if you start it in the foreground the initd never exits.
> command_background="yes". Shizmob has already mentioned it few comments above.

I thought that you’ve started the unbound program in foreground **without** `command_background="yes"` – that’s exactly what would happen in that case.

* * *
I admit that I’ve overlooked your last commit and assumed that it’s still in the same state as before, based on your comments.

Reply:
`-d` in unbound means foreground

```
root@dns2 [~]# unbound -h | grep "\-d"
-d	do not fork into the background.
```